ov/uplink_set.go

```diff
@@ -0,0 +1,236 @@
+
+}
+
+type PortConfigInfos struct {
+	DesiredSpeed string   `json:"desiredSpeed,omitempty"` //"desiredSpeed":"Auto",
+	Location     Location `json:"location"`               //"location":"{...},
+}
+
+type UplinkSet struct {
+	Name                           string            `json:"name,omitempty"`                           // "name": "Uplink77",
+	LogicalInterconnectURI         utils.Nstring     `json:"logicalInterconnectUri,omitempty"`         // "logicalInterconnectUri": "/rest/logical-interconnects/7769cae0-b680-435b-9b87-9b864c81657f",
+	NetworkURIs                    []utils.Nstring   `json:"networkUris,omitempty"`                    // "networkUris": "/rest/uplink-sets/e2f0031b-52bd-4223-9ac1-d91cb519d548",
+	FcNetworkURIs                  []utils.Nstring   `json:"fcNetworkUris"`                            // "fcNetworkUris": "[]",
+	FcoeNetworkURIs                []utils.Nstring   `json:"fcoeNetworkUris"`                          // "fcoeNetworkUris": "[]",
+	PortConfigInfos                []PortConfigInfos `json:"portConfigInfos"`                          // "portConfigInfos": "[]",
+	ConnectionMode                 string            `json:"connectionMode,omitempty"`                 // "connectionMode":"Auto",
+	NetworkType                    string            `json:"networkType,omitempty"`                    // "networkType":"Ethernet",
+	EthernetNetworkType            string            `json:"ethernetNetworkType,omitempty"`            // "ethernetNetworkType":"Tagged",
+	ManualLoginRedistributionState string            `json:"manualLoginRedistributionState,omitempty"` //"manualLoginRedistributionState":"NotSupported"
+	URI                            utils.Nstring     `json:"uri,omitempty"`                            // "uri": "/rest/uplink-sets/"e2f0031b-52bd-4223-9ac1-d91cb519d548",
+	Type                           string            `json:"type,omitempty"`                           // "type": "uplink-setV4",
+}
+
+type UplinkSetList struct {
+	Total       int           `json:"total,omitempty"`       // "total": 1,
+	Count       int           `json:"count,omitempty"`       // "count": 1,
+	Start       int           `json:"start,omitempty"`       // "start": 0,
+	PrevPageURI utils.Nstring `json:"prevPageUri,omitempty"` // "prevPageUri": null,
+	NextPageURI utils.Nstring `json:"nextPageUri,omitempty"` // "nextPageUri": null,
+	URI         string        `json:"uri,omitempty"`         // "uri": "/rest/uplink-sets?start=0&count=10"
+	Members     []UplinkSet   `json:"members,omitempty"`     // "members":[]
+}
+
+func (c *OVClient) GetUplinkSetByName(name string) (UplinkSet, error) {
+	var (
+		upSet UplinkSet
+	)
+	upSets, err := c.GetUplinkSets(fmt.Sprintf("name matches '%s'", name), "name:asc")
+	if upSets.Total > 0 {
+		return upSets.Members[0], err
+	} else {
+		return upSet, err
+	}
+}
+
+func (c *OVClient) GetUplinkSets(filter string, sort string) (UplinkSetList, error) {
```

Comment:
Maybe we can add support for start and count also as it is available as part of api800